pkg/storage/backend-storage/backend-storage.go

```diff
@@ -104,6 +110,119 @@ func PVCForMigrationTarget(pvcStore cache.Store, migration *corev1.VirtualMachin
+	if migration.Status.MigrationState == nil ||
+		migration.Status.MigrationState.TargetPersistentStatePVCName == migration.Status.MigrationState.SourcePersistentStatePVCName {
+		// The migration either didn't actually start, or the backend storage is RWX. Either way we're good, we can delete it.
+		err := client.VirtualMachineInstanceMigration(migration.Namespace).Delete(context.Background(), migration.Name, metav1.DeleteOptions{})
+		return err
+	}
+
+	// An interrupted migration exists. Creating a job to check if the source PVC contains /meta/migrated,
+	// which would indicate that the libvirt migration finished.
+	// A JobComplete condition indicates the file is present, the migration was successful and the target PVC prevails
+	// A JobFailed condition indicated the file is absent, the migration didn't finish and the source PVC prevails
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "backend-storage-recover-",
+		},
+		Spec: batchv1.JobSpec{
+			ActiveDeadlineSeconds:   pointer.P(int64(30)),
+			BackoffLimit:            pointer.P(int32(1)),
+			TTLSecondsAfterFinished: pointer.P(int32(30)),
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "backend-storage-recover-",
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					SecurityContext: &v1.PodSecurityContext{
+						RunAsNonRoot: pointer.P(true),
+						RunAsUser:    pointer.P(int64(util.NonRootUID)),
+						RunAsGroup:   pointer.P(int64(util.NonRootUID)),
+						FSGroup:      pointer.P(int64(util.NonRootUID)),
+						SeccompProfile: &v1.SeccompProfile{
+							Type: v1.SeccompProfileTypeRuntimeDefault,
+						},
+					},
+					Containers: []v1.Container{{
+						Name: "container",
+						SecurityContext: &v1.SecurityContext{
+							AllowPrivilegeEscalation: pointer.P(false),
+							Capabilities:             &v1.Capabilities{Drop: []v1.Capability{"ALL"}},
+						},
+						Image:   launcherImage,
+						Command: []string{"ls"},
+						Args:    []string{"/meta/migrated"},
+						VolumeMounts: []v1.VolumeMount{{
+							Name:      "backend-storage",
+							MountPath: "/meta",
+							SubPath:   "meta",
+						}},
+					}},
+					Volumes: []v1.Volume{{
+						Name: "backend-storage",
+						VolumeSource: v1.VolumeSource{
+							PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+								ClaimName: migration.Status.MigrationState.SourcePersistentStatePVCName,
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+
+	job, err := client.BatchV1().Jobs(vmi.Namespace).Create(context.Background(), job, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Give the job 35 seconds to complete
+	err = virtwait.PollImmediately(time.Second, 35*time.Second, func(ctx context.Context) (done bool, err error) {
```

Comment:
Whatever controller calls it we would block it for up to 35 seconds. Imho this is not ideal and we should make this async.